[DX-810] - Refactor LTS Page (#3731)
TykTechnologies/tyk-docs: tyk-docs/content/frequently-asked-questions/long-term-support-releases.md

---
title: "Long Term Support Releases"

> date: 2023-12-11

Reviewer: @Keithwachira do we keep this? or do we use GH date?

weight: 0
---

Welcome to Tyk's Long Term Support (LTS) Releases page. Here, we'll walk you through the practical aspects of how LTS benefits your business. Explore our approach to stability, understand semantic versioning and learn about our compatibility policies. We'll also cover support for non-LTS components and provide links on upgrading and staying informed about new LTS releases.

> ## Why Customers Love Long Term Support (LTS)?

Reviewer: Are we sure about this tone @dcs3spp ?

> ## Why Customers Love Long Term Support (LTS)?

Reviewer: ```suggestion
## What Makes Long-Term Support (LTS) So Valuable to Our Customers?
```

2. **Security**: Tyk commits that the latest LTS will be secure at the point of release by containing the latest available Go version. This ensures Go related security issues are minimised.
3. **Functional Richness**: There will always be great capability contained in out latest LTS which moves your game on in terms of workflows.
4. **Continuation of Service**: We will patch the LTS version every 7 weeks for the period it remains in full support.

{{< note success >}}
Our current Long Term Support release is version 5.0 LTS, which is in full support until April 2024 and then enters maintenance support until April 2025. Our next Long Term Support Release will be announced end of April 2024.
{{</ note >}}

---

## What Is Our LTS Offering?

We offer a 24 month long term support offering for our Gateway LTS version. For the first 12 months we offer **full support** which means that we will fix every 7 weeks. After this 12 months period is up, we will revert to **maintenance support** for a further 12 months, which offers critical fixes and critical security patching only, with the cadence being on an 'as needed' basis.

> We will release a new LTS Gateway version every 12 months which is upgraded to latest Golang version by default.

Reviewer: `We will release` ? why not `we release`. The future tense suggests it's a new policy but we don't say it's now

> We will release a new LTS Gateway version every 12 months which is upgraded to latest Golang version by default.

Reviewer: ```suggestion
We release a new Gateway LTS version every 12 months which includes the latest Golang version.
```

> Let's just remind ourselves of the benefits to a 12 month LTS window:

Reviewer: Again, not sure about this ToV

> Let's just remind ourselves of the benefits to a 12 month LTS window:

Reviewer: ```suggestion
The advantages of a 12-month Long-Term Support (LTS) window include:"
```

1. It allows us to keep pace with Golang versions and other key dependency upgrades, which kees both our customers and your customers safe (low CVE).
2. It allows us to bring you the best capability whilst offering stability.
3. It allows us to get great product insight and use that to improve the product.

> If you have a requirement to run on a non supported version for longer than the LTS policy, we may be able to help - but this would require a discussion with your Account Manager.

Reviewer: ```suggestion
If you need to operate on a version beyond the LTS policy, there might be options available. Please connect with your Account Manager for a discussion."
```

| Version | Full Support Window | Maintenance Support Window | Completely Unsupported From |
| ---- | ---- | ---- | ---- |
| 4.0 LTS | April 2022 - April 2023 | May 2023 - April 2024 | May 2024 |
| 5.0 LTS | April 2023 - April 2024 | May 2024 - April 2025 | May 2025 |
| LTS+1 (version TBC) | April 2024 - April 2025 | May 2025 - April 2026 | May 2026 |
| LTS+2 (version TBC) | April 2025 - April 2026 | May 2026 - April 2027 | May 2027 |

---

## What About Non LTS Gateway Releases?

> As we release regularly, there will of course be other release issues throughout the year. These are NOT long term support releases. These releases offer new capability and are designed to appeal to teams who want the latest capability as soon as it lands. This capability will be subsumed into the next LTS release, however.

Reviewer: ```suggestion
While we maintain a regular release schedule, it's important to clarify that these releases do not fall under the Long-Term Support (LTS) category. They introduce new capabilities, appealing to teams seeking the latest features upon release. However, these specific features become part of the subsequent LTS release.
```

> These releases are supported, but the support windows is small - they are supported only until the next release comes out and supersedes it. So if you want stabilty and regular patching then LTS releases are for you.

Reviewer: ```suggestion
Although these releases receive support, it's essential to note their limited support duration, extending only until the arrival of the subsequent release that supersedes them. For users prioritizing stability and consistent patching, the LTS releases offer a more suitable choice.
```

---

## Major / Minor / Patch - How Do We Decide?

We know that an LTS release which has a major semantic version is not a desirable practice.

So, we will always endeavour to avoid shipping major versions, especially major versions as LTS releases. However, sometimes it is unavoidable and we have to ship a major version. 

Our first commitment to you is to make our definitions of major / minor / patch transparent:

#### Major Version

The major version is designated as X.0 and is defined by one or more of the following:

1. Breaking changes to Tyk Gateway or Tyk Dashboard APIs. Tyk APIs include not just the endpoints, but also return error codes and messages, logging details - basically anything were a user interacts with a Tyk product and might have to make changes to maintain functionality in response to a change we implement.
2. Deprecation of existing functionality or engines that breaks a key business process.
3. Breaking plugin compiler for customer Go plugins after plugins have been recompiled.
4. Crypto deprecations.
5. Changes to common names in certficates.

> 6. Breaking public function calls.

Reviewer: isn't #6 an example of #1?

Author: Following slack message conversation with @allaneyles have removed list item 6 and amended list item 1 as follows:

_Breaking changes to Tyk Gateway API, Tyk Dashboard API or Tyk custom plugins interfaces. Tyk Gateway and Tyk Dashboard APIs include not just the endpoints, but also return error codes and messages, in addition to logging details. In summary, breaking changes involves anything where a user interacts with a Tyk product and might have to make changes to maintain functionality in response to a change we implement._

7. A change which amends behaviour of a configuration setting.

> Our default setting is NOT to introduce breaking changes, but sometimes it is necessary. If we do introduce breaking changes we have a collective choice to make; if we document the breaking change well and offer migration scripts, we may be able to classify as a minor. If we cannot do that, then it should be a major version.

Reviewer: Not sure I get it. Is it ok to have a small breaking change in a minor? according to semver?

Author: Updated last paragraph of [Major Version](https://deploy-preview-3731--tyk-docs.netlify.app/docs/nightly/frequently-asked-questions/long-term-support-releases/#major-version) section to read:

_In summary, breaking changes involves anything with which a user interacts with a Tyk product and might have to make changes to maintain functionality in response to a change we implement. Subsequently, breaking changes are introduced in a major version, in accordance with the *Semantic Versioning* [specification](https://semver.org/)._

Also mentioned in slack conversation with Allan

Reviewer: sure, are you going to amend the exisiting text?

Author: Done

#### Minor Version

> According the the **Semantic Versioning** specification, a **MINOR** version is incremented when you add functionality in a backward compatible manner. In other words, if Tyk make changes to your software that do not break any existing functionality, you can increment the **MINOR** version number. For example, if you add new features or capabilities to your software without changing any existing functionality, you can increase the **MINOR** version number.

Reviewer: best to add a link to Semantic Versioning

Author: done

> According the the **Semantic Versioning** specification, a **MINOR** version is incremented when you add functionality in a backward compatible manner. In other words, if Tyk make changes to your software that do not break any existing functionality, you can increment the **MINOR** version number. For example, if you add new features or capabilities to your software without changing any existing functionality, you can increase the **MINOR** version number.

Reviewer: ```suggestion
According the the *Semantic Versioning* specification, a MINOR version is incremented when you add functionality in a backward compatible manner. In other words, if Tyk make changes to your software that do not break any existing functionality, you can increment the MINOR version number. For example, if you add new features or capabilities to your software without changing any existing functionality, you can increase the MINOR version number.
```

#### Patch Version

> A patch, sometimes just called a fix, is a small piece of software that's used to correct a problem, usually called a bug, with an operating system or software program.

Reviewer: ```suggestion
A patch, sometimes just called a fix, is a small piece of code that's used to correct a problem, usually called a bug, with an operating system or software program.
```

> Patches are software and Operating System (OS) updates that address security vulnerabilities within a program or product. Tyk may choose to release updates to fix performance bugs, as well as to provide enhanced security features.

Reviewer: Apologies, I don't understand the structure and purpose of the sentence - 
> Patches are software and Operating System (OS) updates that address security vulnerabilities within a program or product.

---

## Compatibility

Tyk has a few different components which can drive questions on what version of X goes with what version of Y.

When we release a new Gateway version, it triggers us to be clear on version compatibility with other areas of the Tyk stack.

As part of the release for the new Gateway LTS version we will commit to show everyone 2 compatibility dimensions:

> 1. What component versions might you need to leverage the very latest Tyk experience across the entire stack.

Reviewer: ```suggestion
1. What component versions would you require to fully utilise the latest Tyk experience across the entire stack
```

2. What components and versions remain backwards compatible with the new Gateway version.

The table below is an illustrative example, but shows the intent:

| Gateway Version | Recommended Compatibility | Backwards Compatibility |
|----    |---- |---- |
| 5.3 LTS | Helm v2.2     | Helm vX - vY |
|         | MDCB v2.5     | MDCB v1.7 - v2.4 |
|         | Operator v1.8 | Operator vX - vY |
|         | Sync v2.4.1   | Sync vX - vY |
| | | EDP vX - vY |
| | | Pump vX - vY |
| | | TIB vX - vY |

> The above compatibility matrix table will be included in future [release notes]({{< ref "product-stack/tyk-gateway/release-notes/overview" >}}) for Gateway for releases 5.3+. Furthermore, the release notes will include tested third party dependences, such as PostgreSQL, MongoDB, Redis etc.

Reviewer: ```suggestion
The compatibility matrix table shown above will be part of upcoming [Gateway release notes]({{< ref "product-stack/tyk-gateway/release-notes/overview" >}}) for versions 5.3 and beyond. Additionally, these release notes will list tested third-party dependencies like *PostgreSQL, MongoDB, Redis*, and more."
```

--- 

## How Do I Upgrade and How Can Tyk Help?

We have step by step install guides for various architectures and installation types.

> Head [here]({{< ref "upgrading-tyk" >}}) to find them!

Reviewer: "here" is not good for SEO. it needs to be part of a sentence of the crawler can understand

> And don't forgeth, our brilliant Customer Success Teams and Account Managers are here to assist you with any issues - pleases refer to your SLA on the specifics of how we can help!

Reviewer: Not sure if it's correct. @vverbani can you please check?

> And don't forgeth, our brilliant Customer Success Teams and Account Managers are here to assist you with any issues - pleases refer to your SLA on the specifics of how we can help!

Reviewer 2: This looks correct @letzya

---

## Keep Me Informed!

> Do you want to know when our next LTS is? And what might be in it? Subscribe to our mailing list [here]()!

Reviewer: ```suggestion
To receive updates on our upcoming Long-Term Support (LTS) release schedule and its contents, feel free to subscribe to our [mailing list]()
```

---

> ## Support Arrangements For Other Tyk Components

Reviewer: we didn't explain for which components the previous content was so this title is not so clear imho

Reviewer: @dcs3spp can you please help to rephrase the title? in the paragraph below we explain that "other" means "not tyk gw or tyk dashboard" but i think the title should be clearer

Author: @letzya I've rephrased the section title to:

_LTS Support Policy For Components Other Than Tyk Gateway and Tyk Dashboard_

#### Enterprise Portal

> We strive to avoid any long term support arrangements for our Enterprise Portal. We run a regular 6 week release cadence which delivers new capability, extension of existing capability and bug fixes. Our poliocy is that we aim to avoid any breaking changes, so in effect the entie Enterprise Portal is supported. Here we'd increment out version as a minor version - 1.3.0, 1.4.0, 1.5.0 etc.

Reviewer: "We strive to avoid any long-term support arrangements" - we can't write this without explaining especially after we explained the opposite beforehand

Reviewer: what does this mean `so in effect the entie Enterprise Portal is supported.`

Author: ```suggested
We strive to avoid any long term support arrangements for our Enterprise Portal. We run a regular 6 week release cadence which delivers new capability, extension of existing capability and bug fixes. Our policy is that we aim to avoid any breaking changes, so in effect the entire Enterprise Portal is supported. Here we'd increment out version as a minor version - 1.3.0, 1.4.0, 1.5.0 etc.
```

> Occasionally, we may see a need to issue a critical fix if there is a systems down time or a criticial security defect. Here would release this as as soon as physically possible and the semantic versioning would reflect a patch (1.3.1, 1.4.1. etc).

Reviewer: should we improve this paragraph? not sure it's read well @dcs3spp

Reviewer: @dcs3spp is this Allan's version?

> Occasionally, we may see a need to issue a critical fix if there is a systems down time or a criticial security defect. Here would release this as as soon as physically possible and the semantic versioning would reflect a patch (1.3.1, 1.4.1. etc).

Author: @letzya 

```suggestion
We strive to avoid any long term support arrangements for our Enterprise Portal. We run a regular 6 week release cadence to deliver new features, extend existing features and address bug fixes. Our policy is that we aim to avoid any breaking changes, so in effect the entire Enterprise Portal is supported. In these cases our version would be released as a minor version, e.g. 1.3.0, 1.4.0, 1.5.0 etc.

Critical fixes and security defects would be fixed and released urgently as a patch release, e.g. 1.3.1, 1.4.1, 1.5.1 etc.
```

> The only exception to this policy is if we every need to release a breaking change. This would mean that we have to release a new major version (i.e. releasing version 2.0). In this exceptional circumstance we would support both the old major version and the new one concurrently for six months - please note that the old version only gets supported in terms of critical fixes, not new functionality. After the six months is up, the previous major version falls out of support.

Reviewer: ```suggestion
The only exception to this policy is if we ever need to release a breaking change. This would mean that we have to release a new major version (i.e. releasing version 2.0). In this exceptional circumstance we would support both the old major version and the new one concurrently for six months - please note that the old version only gets supported in terms of critical fixes, not new functionality. After six months, the previous major version falls out of support.
```

Reviewer: are we sure we can write "exceptional circumstance" at this stage?

> If you are using the Enterprise Portal on Tyk Cloud, we recommend you use the latest version at all times and we auto upgrade to that version on TYk Cloud. When a new release is launched, older versions are immediately deprecated . This gives you the best, latest and most secure capability. If a customer wants to use a previous version, this is possible via rollback; but if an issue arises on an older version our advice would be simply to upgrade to the latest version.

Reviewer: I don't quite understand this. "we recommend you use the latest" but we auto upgrade? is it because the user can manually downgrade after we upgraded them automatically?

> #### Other Components

Reviewer: ```suggestion
#### Tyk Pump, Tyk Identity Broker, MDCB and Tyk Operator
```

> Minor updates to Tyk Pump, Tyk Identity Broker (TIB), MDCB and Operator are deployable quickly with low risk and no breaking changes. Tyk will support the latest major.minor version (e.g. Pump 1.7) until the next major.minor version (1.8) is released. Then we'd support that version.

Reviewer: ```suggestion
Minor updates to Tyk Pump, Tyk Identity Broker (TIB), MDCB and Tyk Operator are deployable swiftly with minimal risk and no breaking changes. Tyk will provide support for the latest *major.minor* version (e.g. Tyk Pump v1.7) until the subsequent *major.minor* version (e.g. Tyk Pump v1.8) is released, at which point support for the newer version will commence."
```

> To help assure backward compatibility we ensure that each version of Pump and other components we release works with the Gateway and Dashboard versions which are under Long Term Support (LTS) at that time. For example, we'd ensure Pump v1.7 is compatible with release v4 of Tyk Gateway and Tyk Dashboard, and this increments with the LTS model.Following semver convention, if we need to patch a minor Pump version, we would number that as a patch version (e.g. 1.7.1).

Reviewer: ```suggestion
To ensure backward compatibility, we confirm that each Pump release and other components are compatible with the Long-Term Support (LTS) versions of the Gateway and Dashboard. For instance, *Tyk Pump* v1.7 undergoes validation to ensure compatibility with release v4 of *Tyk Gateway* and *Tyk Dashboard*. This alignment persists under the LTS model. Following the semantic versioning convention, if there's a need for a fix in the *Tyk Pump* version, it will be labelled as a patch version (e.g., v1.7.1).
```

#### Enterprise Portal

We strive to avoid any long term support arrangements for our Enterprise Portal. We run a regular 6 week release cadence which delivers new capability, extension of existing capability and bug fixes. Our policy is that we aim to avoid any breaking changes, so in effect the entire Enterprise Portal is supported. Here we'd increment out version as a minor version - 1.3.0, 1.4.0, 1.5.0 etc.

Occasionally, we may see a need to issue a critical fix if there is a systems down time or a criticial security defect. Here would release this as as soon as physically possible and the semantic versioning would reflect a patch (1.3.1, 1.4.1. etc).

The only exception to this policy is if we every need to release a breaking change. This would mean that we have to release a new major version (i.e. releasing version 2.0). In this exceptional circumstance we would support both the old major version and the new one concurrently for six months - please note that the old version only gets supported in terms of critical fixes, not new functionality. After the six months is up, the previous major version falls out of support.

> If you are using the Enterprise Portal on Tyk Cloud, we recommend you use the latest version at all times and we auto upgrade to that version on TYk Cloud. When a new release is launched, older versions are immediately deprecated . This gives you the best, latest and most secure capability. If a customer wants to use a previous version, this is possible via rollback; but if an issue arises on an older version our advice would be simply to upgrade to the latest version.

Author: @letzya revised comment from @allaneyles via slack message
```suggestion
If you are using Enterprise Portal on Tyk Cloud, we keep you safe, secure and up to date with the latest features by auto upgrading you to the latest version. When a new release is launched, older versions are immediately deprecated. If a customer wants to use a previous version, this is possible via rollback. However, if an issue arises on an older version our advice would be to upgrade to the latest version.
```

Reviewer: much better

---

## Major / Minor / Patch - How Do We Decide?

We know that an LTS release which has a major semantic version is not a desirable practice.

So, we will always endeavour to avoid shipping major versions, especially major versions as LTS releases. However, sometimes it is unavoidable and we have to ship a major version. 

Our first commitment to you is to make our definitions of major / minor / patch transparent:

#### Major Version

The major version is designated as X.0 and is defined by one or more of the following:

1. Breaking changes to Tyk Gateway API, Tyk Dashboard API or Tyk custom plugins interfaces. Tyk Gateway and Tyk Dashboard APIs include not just the endpoints, but also return error codes and messages, in addition to logging details. In summary, breaking changes involves anything where a user interacts with a Tyk product and might have to make changes to maintain functionality in response to a change we implement.
2. Deprecation of existing functionality or engines that breaks a key business process.

> 3. Breaking plugin compiler for customer Go plugins after plugins have been recompiled.

Reviewer: ```suggestion
3. Breaking changes in the Go plugin compiler for users' Go plugins after plugins have been recompiled.
```

> 4. Crypto deprecations.

Reviewer: can we move this next to the other deprecation item #2

Author: What is the other deprecation item and does it relate to this PR?

Reviewer: i wrote it under #4 and asked to put it near #2

Author: Updated breaking changes list as follows:

_1. Breaking changes to Tyk APIs, including Tyk Gateway API, Tyk Dashboard API, MDCB and Tyk EDP and any other component which exposes APIs. Changes include not just the endpoints, but also behaviour and functionality, schemas, input parameters, return error codes and messages. The APIs are documented and published as an OpenAPI Spec document.
2. Breaking changes to Tyk custom plugins interfaces, breaking plugin compiler for customer Go plugins after plugins have been recompiled.
3. Breaking changes in the config files in all Tyk components, fields in the config files, environment variables used by Tyk components, APIs (function calls) of Tyk middleware, Go template interface.
4. Breaking changes in the Tyk Gateway and Tyk Dashboard log messages.
5. Deprecation of existing functionality or engines that breaks a key business process.
6. Crypto deprecations.
7. Changes to common names in certfiicates._

#### Patch Version

A patch, sometimes just called a fix, is a small piece of code that's used to correct a problem, usually called a bug, with an operating system or software program.

Patches are software and Operating System (OS) updates that address security vulnerabilities within a program or product. Tyk may choose to release updates to fix performance bugs, as well as to provide enhanced security features.

---

## Compatibility

Tyk has a few different components which can drive questions on what version of X goes with what version of Y.

When we release a new Gateway version, it triggers us to be clear on version compatibility with other areas of the Tyk stack.

As part of the release for the new Gateway LTS version we will commit to show everyone 2 compatibility dimensions:

> 1. What component versions would you require to fully utilise the latest Tyk experience across the entire stack

Reviewer: ```suggestion
1. **Recommended releases** - To ensure you get the most out of the latest Tyk experience, we'll provide information on which versions of different components across the entire stack you need.
```

> 2. What components and versions remain backwards compatible with the new Gateway version.

Reviewer: ```suggestion
2. **Backwards Compatibility** - We'll provide information on what components and versions remain backwards compatible with the new Tyk Dashboard and Tyk Gateway versions.
```

The table below is an illustrative example, but shows the intent:

> | Gateway Version | Recommended Compatibility | Backwards Compatibility |

Reviewer: ```suggestion
| Gateway Version | Recommended releases | Backwards Compatibility |
```

We strive to avoid any long term support arrangements for our Enterprise Portal. We run a regular 6 week release cadence which delivers new capability, extension of existing capability and bug fixes. Our policy is that we aim to avoid any breaking changes, so in effect the entire Enterprise Portal is supported. Here we'd increment out version as a minor version - 1.3.0, 1.4.0, 1.5.0 etc.

We support our release cadence with a critical fix process. This is invoked when a severity one incident is declared, usually defined as systems down time, no workaround or a critical security issue.

Our critical fix process commits to release as soon as possible and we issue a standalone fix outside of our normal cadence.

The only exception to this policy is if we every need to release a breaking change. This would mean that we have to release a new major version (i.e. releasing version 2.0). In this exceptional circumstance we would support both the old major version and the new one concurrently for six months - please note that the old version only gets supported in terms of critical fixes, not new functionality. After the six months is up, the previous major version falls out of support.

> If you are using the Enterprise Portal on Tyk Cloud, we recommend you use the latest version at all times and we auto upgrade to that version on TYk Cloud. When a new release is launched, older versions are immediately deprecated . This gives you the best, latest and most secure capability. If a customer wants to use a previous version, this is possible via rollback; but if an issue arises on an older version our advice would be simply to upgrade to the latest version.

Reviewer: ```suggestion
I don't see much wrong with the sentence. But we can change to 'If you are using Enterprise Portal on Tyk Cloud, we keep you safe, secure, and up to date with the latest features by auto-upgrading you to the latest version. When a new release is launched, older versions are immediately deprecated. If a customer wants to use a previous version, this is possible via rollback; but if an issue arises on an older version our advice would be simply to upgrade to the latest version.
```

2. **Security**: Tyk commits that the latest LTS will be secure at the point of release by containing the latest available Go version. This ensures Go related security issues are minimised.
3. **Functional Richness**: There will always be great capability contained in out latest LTS which moves your game on in terms of workflows.
4. **Continuation of Service**: We will patch the LTS version every 7 weeks for the period it remains in full support.

In summary, LTS releases are stable minor or patch releases that are suitable for production use.

{{< note success >}}
Our current Long Term Support release is version 5.0 LTS, which is in full support until April 2024 and then enters maintenance support until April 2025. Our next Long Term Support Release will be announced end of April 2024.
{{</ note >}}

---

## What Is Our LTS Offering?

We provide full support for the first 12 months, including regular maintenance intervals every seven weeks. Following this period, we transition to maintenance support for an additional 12 months, focusing on critical fixes and essential security patching as needed.

> We offer a 24 month long term support offering for our Gateway LTS version. For the first 12 months we offer **full support** which means that we will fix every 7 weeks. After this 12 months period is up, we will revert to **maintenance support** for a further 12 months, which offers critical fixes and critical security patching only, with the cadence being on an 'as needed' basis.

Reviewer 3: ```suggestion
We offer a 24 month long term support offering for our Gateway LTS version. For the first 12 months we offer **full support** which means that we will fix every 7 weeks. After this 12 month period is up, we will revert to **maintenance support** for a further 12 months offering critical fixes and critical security patching only, with the cadence being on an 'as needed' basis.
```

We release a new Gateway LTS version every 12 months which includes the [latest stable Golang version](https://go.dev/dl/).

The advantages of a 12-month Long-Term Support (LTS) window include:

> 1. It allows us to keep pace with Golang versions and other key dependency upgrades, which kees both our customers and your customers safe (low CVE).

Reviewer 3: ```suggestion
1. It allows us to keep pace with Golang versions and other key dependency upgrades, which keeps both our customers and your customers safe (low CVE).
```

2. It allows us to bring you the best capability whilst offering stability.
3. It allows us to get great product insight and use that to improve the product.

> In case there's a need to operate on a version beyond the established LTS policy, potential assistance may be available. However, this requires a discussion with your designated Account Manager."

Reviewer 3: ```suggestion
In case there's a need to operate on a version beyond the established LTS policy, potential assistance may be available. However, this requires a discussion with your designated Account Manager.
```

> | Version | Full Support Window | Maintenance Support Window | Completely Unsupported From |

Reviewer 3: "Completely unsupported" has a feeling of danger to it - how about "End of Life" or "Support Ends"

| ---- | ---- | ---- | ---- |
| 4.0 LTS | April 2022 - April 2023 | May 2023 - April 2024 | May 2024 |
| 5.0 LTS | April 2023 - April 2024 | May 2024 - April 2025 | May 2025 |
| LTS+1 (version TBC) | April 2024 - April 2025 | May 2025 - April 2026 | May 2026 |
| LTS+2 (version TBC) | April 2025 - April 2026 | May 2026 - April 2027 | May 2027 |

---

## What About Non LTS Gateway Releases?

> While we maintain a regular release schedule, it's important to clarify that these releases do not fall under the Long-Term Support (LTS) category. They introduce new capabilities, appealing to teams seeking the latest features upon release. However, these specific features become part of the subsequent LTS release.

Reviewer 3: ```suggestion
While we maintain a regular release schedule, it's important to clarify that these releases do not fall under the Long Term Support (LTS) category. They introduce new capabilities, appealing to teams seeking the latest features upon release. However, these specific features become part of the subsequent LTS release.
```

#### Minor Version

According to the *Semantic Versioning* [specification](https://semver.org/), a MINOR version is incremented when you add functionality in a backwards compatible manner. In other words, if Tyk makes changes to your software that do not break any existing functionality, you can increment the MINOR version number. For example, if you add new features or capabilities to your software without changing any existing functionality, you can increase the MINOR version number.

#### Patch Version

A patch, sometimes just called a fix, is a small piece of code that's used to correct a problem, usually called a bug, with an operating system or software program.

Patches are software and Operating System (OS) updates that address security vulnerabilities within a program or product. Tyk may choose to release updates to fix performance bugs, as well as to provide enhanced security features.

---

## Compatibility

Tyk has a few different components which can drive questions on what version of X goes with what version of Y.

When we release a new Gateway version, it triggers us to be clear on version compatibility with other areas of the Tyk stack.

> As part of the release for the new Gateway LTS version we will commit to show everyone 2 compatibility dimensions:

Reviewer 3: ```suggestion
As part of the release for the new Gateway LTS version we will commit to show everyone two compatibility dimensions:
```

|         | Operator v1.8 | Operator v0.16 |
|         | Sync v2.4.1   | Sync vX - vY |
| | EDP v1.8.3 | EDP all versions |
| | Pump v1.9.0 | Pump all versions |
| | TIB (if using standalone) v1.5 | TIB all versions |

The compatibility matrix table shown above will be part of upcoming [Gateway release notes]({{< ref "product-stack/tyk-gateway/release-notes/overview" >}}) for versions 5.3 and beyond. Additionally, these release notes will list tested third-party dependencies like *PostgreSQL, MongoDB, Redis*, and more.

--- 

## How Do I Upgrade and How Can Tyk Help?

We have step by step install guides for various architectures and installation types. Refer to [upgrading tyk]({{< ref "upgrading-tyk" >}}) for further details.

And don't forget, our brilliant Customer Success Teams and Account Managers are here to assist you with any issues - pleases refer to your SLA on the specifics of how we can help!

---

## Keep Me Informed!

To receive updates on our upcoming Long-Term Support (LTS) release schedule and its contents, feel free to subscribe to our [mailing list]()

---

## LTS Support Policy For Components Other Than Tyk Gateway and Tyk Dashboard

> The Tyk product is comprised of various components, such as Tyk Sync, Tyk Pump and Tyk Operator. Some of these components are more standalone than others, which allows us to operate different release cadences. Consequently, we strive to avoid long term support policies for any component, except Tyk Gateway and Tyk Dashboard.

Reviewer 3: Wouldn't it be simpler just to say "we do not have a long term support policy"?

"Striving to avoid" means that, while we aim to avoid this, sometimes we might fail... which would mean (incorrectly) that there ARE long term support versions from time to time...

Reviewer: @allaneyles ?

Author: @letzya @andyo-tyk

@allaneyles Has updated the support for [other Tyk components ](https://deploy-preview-4051--tyk-docs.netlify.app/docs/nightly/frequently-asked-questions/long-term-support-releases/#support-arrangements-for-other-tyk-components)

Tyk Gateway and Tyk Dashboard represent the core product and contain the majority of our workflows. Subsequently, these have a long term support policy.

#### Enterprise Portal

> We strive to avoid any long term support arrangements for our Enterprise Portal. We run a regular 6 week release cadence which delivers new capability, extension of existing capability and bug fixes. Our policy is that we aim to avoid any breaking changes, so in effect the entire Enterprise Portal is supported. Here we'd increment out version as a minor version - 1.3.0, 1.4.0, 1.5.0 etc.

Reviewer 3: Same here for "strive to avoid" - I don't think it makes sense.

Author: @andyo-tyk

@allaneyles Has updated the support for [other Tyk components ](https://deploy-preview-4051--tyk-docs.netlify.app/docs/nightly/frequently-asked-questions/long-term-support-releases/#support-arrangements-for-other-tyk-components) Enterprise portal section is now removed.

We support our release cadence with a critical fix process. This is invoked when a severity one incident is declared, usually defined as systems down time, no workaround or a critical security issue.

Our critical fix process commits to release as soon as possible and we issue a standalone fix outside of our normal cadence.

> The only exception to this policy is if we ever need to release a breaking change. This would mean that we have to release a new major version (i.e. releasing version 2.0). In this exceptional circumstance we would support both the old major version and the new one concurrently for six months - please note that the old version only gets supported in terms of critical fixes, not new functionality. After six months, the previous major version falls out of support.

Reviewer 3: ```suggestion
The only exception to this policy is if we ever need to release a breaking change. This would mean that we have to release a new major version (e.g. version 2.0). In this exceptional circumstance we would support both the old major version and the new one concurrently for six months - please note that the old version would only be supported in terms of critical fixes, not new functionality. After six months, the previous major version would fall out of support.
```

If you are using Enterprise Portal on Tyk Cloud, we keep you safe, secure and up to date with the latest features by auto upgrading you to the latest version. When a new release is launched, older versions are immediately deprecated. If a customer wants to use a previous version, this is possible via rollback. However, if an issue arises on an older version our advice would be to upgrade to the latest version.

#### Tyk Pump, Tyk Identity Broker, MDCB and Tyk Operator

> Minor updates to Tyk Pump, Tyk Identity Broker (TIB), MDCB and Tyk Operator are deployable swiftly with minimal risk and no breaking changes. Tyk will provide support for the latest *major.minor* version (e.g. Tyk Pump v1.7) until the subsequent *major.minor* version (e.g. Tyk Pump v1.8) is released, at which point support for the newer version will commence."

Reviewer 3: ```suggestion
Minor updates to Tyk Pump, Tyk Identity Broker (TIB), MDCB and Tyk Operator are deployable swiftly with minimal risk and no breaking changes. Tyk will provide support for the latest *major.minor* version (e.g. Tyk Pump v1.7) until the subsequent *major.minor* version (e.g. Tyk Pump v1.8) is released, at which point support for the newer version will commence.
```

> To ensure backward compatibility, we confirm that each Pump release and other components are compatible with the Long-Term Support (LTS) versions of the Gateway and Dashboard. For instance, *Tyk Pump* v1.7 undergoes validation to ensure compatibility with release v4 of *Tyk Gateway* and *Tyk Dashboard*. This alignment persists under the LTS model. Following the semantic versioning convention, if there's a need for a fix in the *Tyk Pump* version, it will be labelled as a patch version (e.g., v1.7.1).

Reviewer 3: ```suggestion
To ensure backward compatibility, we confirm that each Pump release and other components are compatible with the Long Term Support (LTS) versions of the Gateway and Dashboard. For instance, *Tyk Pump* v1.8 underwent validation to ensure compatibility with release v5 of *Tyk Gateway* and *Tyk Dashboard*. This alignment persists under the LTS model. Following the semantic versioning convention, if there's a need for a fix in the *Tyk Pump* version, it would be labelled as a patch version (e.g. v1.8.1).
```
